src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py

```diff
@@ -1025,10 +1035,28 @@ def _calculate_curve_number_raster(
             numpy.array(lulc_to_soil[soil_id]['cn_values'],
                         dtype=numpy.float32))
 
+    # Use set of table lucodes in cn_op
+    lucodes_set = set(list(biophysical_table))
+
     def cn_op(lulc_array, soil_group_array):
         """Map lulc code and soil to a curve number."""
         cn_result = numpy.empty(lulc_array.shape)
         cn_result[:] = cn_nodata
+
+        # if lulc_array value not in lulc_to_soil[soil_group_id]['lulc_values']
+        # then numpy.digitize will not bin properly and cause an IndexError
+        # during the reshaping call
+        lulc_unique = set(numpy.unique(lulc_array))
+        if not lulc_unique.issubset(lucodes_set):
+            # cast to list to conform with similar error messages in InVEST
+            missing_lulc_values = sorted(lulc_unique.difference(lucodes_set))
+            error_message = (
+                "Values in the LULC raster were found that are not"
+                " represented under the 'lucode' key column of the"
```

Comment:
@phargogh : I would be happy to drop it except that the *default* message returned from `utils.reclassify_raster` when no `error_details` are provided is: 
>Values in the raster were found that are not represented under the key column of the...

Notice `LULC` and `'lucode'` are missing in the default case where no details were provided. So I was trying to be consistent here (I understand it's not use the utils func) to match what `utils.reclassify_raster` would provide for a message. I purposefully placed `raster` and `key` in `utils.reclassify_raster` so that the message would still read appropriately. 

But perhaps this lends to the case of whether that `error_details` input should even be optional. This has me leaning towards no, it should be required.

Thoughts?

Reply:
In responding to your other comment below I also am thinking that maybe a better error message that would drop `key` and add `corresponding` would be:
```
  "Values in the{formatted_details['raster_name']}raster were"
                " found that are not represented under the CORRESPONDING"
                f"{formatted_details['column_name']} column of the"
                f"{formatted_details['table_name']}table. The missing values"
                f" found in the{formatted_details['raster_name']}raster but"
                f" not the table are: {err.missing_values}.")
```

Reply:
@dcdenu4 If I can summarize from various threads here, it's starting to sound like we all think it might be a good idea to require all the error details. And then it's a bonus that this message formatting will get simpler if it's not trying to handle both the presence & absence of those extra words!

Reply:
`corresponding` would work well as an alternative to `key`, and I agree with Dave that it seems like we've reached consensus about requiring the error details, which would also make much of the formatting a bit simpler.